src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -148,6 +149,9 @@ protected FileMonitoringController(FilePath ws) throws IOException, InterruptedE
             if (status.exists()) {
                 try {
                     return Integer.parseInt(status.readToString().trim());
+                } catch (FileNotFoundException ignored) {
+                    /* File is locked. Ignore and just try again later. */
```

Comment:
I have not dug into the backing code of the `readToString` function. In case the file was locked the `exists` function returned `true` and the `readToString` function did throw that exception.